File: tools/reveal.cpp
// This is a reveal program. It reveals the map.

#include <iostream>
#include <integers.h>
#include <vector>
using namespace std;

#include <DFTypes.h>
#include <DFHackAPI.h>

int main (void)
{
    uint32_t x_max,y_max,z_max;
    DFHack::designations40d designations;
    
    DFHack::API DF("Memory.xml");
    if(!DF.Attach())
    {
        cerr << "DF not found" << endl;
        return 1;
    }
    DF.InitMap();
    DF.getSize(x_max,y_max,z_max);
    
    // walk the map
    for(uint32_t x = 0; x< x_max;x++)
    {
        for(uint32_t y = 0; y< y_max;y++)
        {
            for(uint32_t z = 0; z< z_max;z++)
            {
                if(DF.isValidBlock(x,y,z))
                {
                    // read block designations
                    DF.ReadDesignations(x,y,z, &designations);
                    // change the hidden flag to 0
                    for (uint32_t i = 0; i < 16;i++) for (uint32_t j = 0; j < 16;j++)
                    {
                        designations[i][j].bits.hidden = 0;
                    }
                    // write the designations back
                    DF.WriteDesignations(x,y,z, &designations);
                }
            }
        }
    }
    DF.Detach();
    #ifndef LINUX_BUILD
    cout << "Done. Press any key to continue" << endl;
    cin.ignore();
    #endif
    return 0;
}

File: tools/python/reveal.py
from pydfhack import ContextManager, Maps

df_cm = ContextManager("Memory.xml")
df = df_cm.get_single_context()

df.attach()

m = df.maps()

m.start()

m_x, m_y, m_z = m.size

for x in xrange(m_x):
    for y in xrange(m_y):
        for z in xrange(m_z):
            if m.is_valid_block(x, y, z):
                d = m.read_designations(x, y, z)
                
                for i in d:
                    for j in i:
                        j.bits.hidden = 0
                
                m.write_designations(x, y, z, d)

m.finish()
df.detach()